cmake/modules/RootBuildOptions.cmake

```
@@ -406,6 +404,15 @@ foreach(opt )
   endif()
 endforeach()
 
+if(DEFINED rpath)
+  message(DEPRECATION ">>> Option 'rpath' is deprecated and is ignored."
+      " Relative RPATHs to the main ROOT libraries are unconditionally appended to all ROOT"
+      " executables and libraries. The 'rpath' flag presence will result in a configuration"
```

Here, I'm missing the explanation that it's not needed (because CMake is anyway setting rpaths at build and install time). Would linking to the release notes be an option?

Furthermore, forcing a build error might be annoying for people who build different ROOT versions, because they always have to flip the switch on and off. Wouldn't the deprecation message be sufficient?
```suggestion
  message(DEPRECATION ">>> Option 'rpath' is deprecated and without effect."
      " See https://root.cern/doc/v638/release-notes.html"
      " Setting the 'rpath' flag will result in a configuration"
```